cpg-console/src/main/kotlin/de/fraunhofer/aisec/cpg/console/CompilationDatabase.kt

```diff
@@ -0,0 +1,110 @@
+        override val short: String by conf.get(default = "trdb")
+        override val description: String =
+            "translates the source code files using the provided compilation database into the CPG"
+
+        override val params = "<path>"
+
+        override fun execute(line: String): Command.Result {
+            val p = line.indexOf(' ')
+            val path = line.substring(p + 1).trim()
+
+            return Command.Result.RunSnippets(
+                listOf(
+                    // basics
+                    "import de.fraunhofer.aisec.cpg.TranslationConfiguration",
+                    "import de.fraunhofer.aisec.cpg.TranslationManager",
+                    // go
+                    "import de.fraunhofer.aisec.cpg.frontends.golang.GoLanguageFrontend",
+                    // llvm
+                    "import de.fraunhofer.aisec.cpg.frontends.llvm.LLVMIRLanguageFrontend",
+                    // python
+                    "import de.fraunhofer.aisec.cpg.frontends.python.PythonLanguageFrontend",
+                    // all the graph nodes
+                    "import de.fraunhofer.aisec.cpg.graph.*",
+                    "import de.fraunhofer.aisec.cpg.graph.declarations.*",
+                    "import de.fraunhofer.aisec.cpg.graph.statements.*",
+                    "import de.fraunhofer.aisec.cpg.graph.statements.expressions.*",
+                    // helper builtins
+                    "import de.fraunhofer.aisec.cpg.graph.evaluate",
+                    "import de.fraunhofer.aisec.cpg.graph.all",
+                    "import de.fraunhofer.aisec.cpg.graph.ast",
+                    "import de.fraunhofer.aisec.cpg.graph.dfgFrom",
+                    "import de.fraunhofer.aisec.cpg.graph.byName",
+                    "import de.fraunhofer.aisec.cpg.graph.body",
+                    "import de.fraunhofer.aisec.cpg.graph.capacity",
+                    "import de.fraunhofer.aisec.cpg.console.printCode",
+                    // some basic java stuff
+                    "import java.io.File",
+                    // prepare the compilation database
+                    "import de.fraunhofer.aisec.cpg.frontends.CompilationDatabase.Companion.fromFile",
+                    "val db = fromFile(File(\"$path\"))",
+                    // lets build and analyze
+                    "@OptIn(de.fraunhofer.aisec.cpg.ExperimentalGolang::class, de.fraunhofer.aisec.cpg.ExperimentalPython::class) val config =\n" +
+                        "                TranslationConfiguration.builder()\n" +
+                        "                    .useCompilationDatabase(db)\n" +
+                        "                    .sourceLocations(db.sourceFiles)\n" +
+                        "                    .defaultLanguages()\n" +
+                        //                      "
+                        // .registerLanguage(GoLanguageFrontend::class.java,
```

Comment:
Any particular reason not to include the go frontend? Do we also exclude it on `:tr`?

Reply:
Actually, when I think of it, isn't the C++ framework the only one who actually supports a translation database? So we can probably exclude all the other frontends?

Reply:
No particular reason. I just copied everything from the `:tr` plugin...